Document the new `onServerInit` hook (#911)
front-commerce/developers.front-commerce.com: versioned_docs/version-2.x/advanced/server/add-http-endpoint.mdx

  },
};
```

## Extend the HTTP server with `onServerInit`

<SinceVersion tag="2.30" />

> The `onServerInit` hook is a server module extension point introduced in
> Front-Commerce 2.30. It allows you to extend the Express HTTP server or perform
> additional tasks during server initialization.

Reviewer: nitpick: the `SinceVersion` already explains that it's introduced in `Front-Commerce 2.30` I don't think it's necessary to repeat it.

 ```suggestion
The `onServerInit` hook allows you to extend the Express HTTP server or perform
additional tasks during server initialization.
```

Author: Updated with your suggestion

To use the `onServerInit` hook, follow these steps:

> 1. Create a new server module or open an existing one.

Reviewer: It could be good to link to the server module docs.

```suggestion
1. Create a new [server module](/docs/2.x/advanced/server/add-http-endpoint#what-is-a-module) or open an existing one.
```

Author: Updated with your suggestion

> 1. Create a new server module or open an existing one.
> 2. In the `module.config.js` file of your server module, define the
>    `onServerInit` function. This function will be called when the server is
>    initialized and will receive an instance of the current server object as a
>    parameter.
> 
> ```js title="my-module/server/module.config.js"
> export default {
>   onServerInit: (server) => {
>     // Perform server initialization tasks here
>     console.log("Server is initialized");
>   },
> };
> ```
> 
> 3. Save the changes to your `module.config.js` file.

Reviewer: These steps seem a bit redundant. 

For example `Save the changes to your `module.config.js` file.` brings no value.
and `Create a new server module or open an existing one.` is specified in step `2`

```suggestion
Create a new [server module](/docs/2.x/advanced/server/add-http-endpoint#what-is-a-module) 
or open an existing one. In the `module.config.js` file of your server module, define the
`onServerInit` function. This function will be called when the server is
initialized and will receive an instance of the current server object as a
parameter.

```js title="my-module/server/module.config.js"
export default {
  onServerInit: (server) => {
    // Perform server initialization tasks here
    console.log("Server is initialized");
  },
};
```

You should now see a `"Server is initialized"` console log once your server has started.
```

Author: I adapted the content with a slightly different one than suggested, but removed duplication and the useless bullet points.

:::tip

We have created a working example of a server module that uses the
`onServerInit` hook to add a WebSocket server to the Express server. You can
find the example in the

> [Front-Commerce examples repository](https://github.com/front-commerce/examples).

Reviewer: Could be better to directly link to the example, once it's merged

Author: Link updated to https://github.com/front-commerce/examples/tree/main/websocket-server
